src/toolbox/utils/color/color.spec.ts

```diff
@@ -1,52 +1,103 @@
-      ['rgb(255,105,180)', [255,105,180,1]],
-      ['rgba(255,105,180,1)', [255,105,180,1]],
-      ['rgba(255,105,180,0.5)', [255,105,180,.5]],
-      ['rgba(255,105,180,.5)', [255,105,180,.5]],
-      ['#FFF', [255,255,255,1]],
+      ["red", [255, 0, 0, 1]],
+      ["hotpink", [255, 105, 180, 1]],
+      ["#ff69b4", [255, 105, 180, 1]],
+      ["rgb(255,105,180)", [255, 105, 180, 1]],
+      ["rgba(255,105,180,1)", [255, 105, 180, 1]],
+      ["rgba(255,105,180,0.5)", [255, 105, 180, 0.5]],
+      ["rgba(255,105,180,.5)", [255, 105, 180, 0.5]],
+      ["#FFF", [255, 255, 255, 1]]
     ];
     tests.forEach(([testInput, expectedResult]) => {
-      it(
-        `should return ${JSON.stringify(expectedResult)} from toNumbers when
-         constructed with Color.fromStyleString(${JSON.stringify(testInput)})`,
-        () => {
-          const result = Color.fromStyleString(<string>testInput).toNumbers();
-          expect(result).to.have.ordered.members(<number[]>expectedResult);
-          expect(result.length).to.equal((<number[]>expectedResult).length)
-        }
-      );
+      it(`should return ${JSON.stringify(expectedResult)} from toNumbers when
+         constructed with Color.fromStyleString(${JSON.stringify(
+           testInput
+         )})`, () => {
+        const result = Color.fromStyleString(<string>testInput).toNumbers();
+        expect(result).to.have.ordered.members(<number[]>expectedResult);
+        expect(result.length).to.equal((<number[]>expectedResult).length);
+      });
+    });
+  });
+
+  describe("fromHex", () => {
+    const tests = [
+      ["#FFFFFF", [255, 255, 255, 1]],
+      ["#000000", [0, 0, 0, 1]],
+      ["#812dd3", [129, 45, 211, 1]],
+      ["#F2F", [255, 34, 255, 1]],
+      ["FFFFFF", [255, 255, 255, 1]]
+    ];
+    tests.forEach(([testInput, expectedResult]) => {
+      it(`should return ${JSON.stringify(
+        expectedResult
+      )} with Color.fromHex(${JSON.stringify(testInput)})`, () => {
+        const result = Color.fromHex(<string>testInput).toNumbers();
+        expect(areArrayValuesEqual(result, <number[]>expectedResult)).to.equal(
+          true
+        );
+      });
+    });
+  });
+
+  describe("fromRgb", () => {
+    const tests = [
+      ["255, 255, 255", [255, 255, 255, 1]],
+      ["0, 0, 0", [0, 0, 0, 1]],
+      ["129, 45, 211", [129, 45, 211, 1]],
+      ["255, 34, 255", [255, 34, 255, 1]]
+    ];
+    tests.forEach(([testInput, expectedResult]) => {
+      it(`should return ${JSON.stringify(
+        expectedResult
+      )} with Color.fromRgb(${JSON.stringify(testInput)})`, () => {
+        const result = Color.fromRgb(<string>testInput).toNumbers();
+        expect(areArrayValuesEqual(result, <number[]>expectedResult)).to.equal(
+          true
+        );
+      });
+    });
+  });
+
+  describe("isHexValue", () => {
+    const tests = [
+      ["#F2F2F2", true],
+      ["#FFF", true],
+      ["000000", true],
+      ["FFF", true],
+      ["cat", false]
+    ];
```

Comment:
How does this handle valid hex values that aren't 3 or six characters?
FFFF? FF? FFFFF? FFFFFFF?